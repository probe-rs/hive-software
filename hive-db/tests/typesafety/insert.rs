use hive_db::{CborDb, HiveDb, Key};

const FLUSH_INTERVAL_MS: u64 = 60_000;
const CACHE_CAPACITY: u64 = 52_428_800;

fn main() {
    let db = HiveDb::open("dummy_path", FLUSH_INTERVAL_MS, CACHE_CAPACITY);

    let tree = db.open_tree("tree");

    let my_key: Key<u8> = Key::new("NiceNumber".to_owned());

    let wrong_type = "42";

    tree.c_insert(&my_key, wrong_type).unwrap();
}
