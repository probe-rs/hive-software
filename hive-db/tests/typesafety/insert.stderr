error[E0308]: mismatched types
  --> tests/typesafety/insert.rs:15:28
   |
15 |     tree.c_insert(&my_key, wrong_type).unwrap();
   |                            ^^^^^^^^^^ expected `u8`, found `str`
   |
   = note: expected reference `&u8`
              found reference `&str`
